tests/MongoDB.Driver.Core.Tests/Core/Operations/BulkMixedWriteOperationTests.cs

```diff
@@ -1286,9 +1286,11 @@ public void WriteConcern_should_work()
             [Values(false, true)] bool async)
         {
             RequireServer.Check();
-            DropCollection();
+
+            var collectionNamespace = CoreTestConfiguration.GetCollectionNamespaceForTestMethod(nameof(BulkMixedWriteOperationTests), nameof(Execute_with_insert_should_not_send_session_id_when_unacknowledged_writes));
```

Comment:
It looks like the problem is related to how `Unacknowledged` writes is processed by the server.
After this test, there is a possible situation that the inserted record from this test won't be fully processed. And then, the record from this test can be considered as part of a different test that will fail the next test.

Probably these lines fix is the simplest solution to fix this test.

Reply:
You should make the same change for these other methods also:

    Execute_with_delete_should_not_send_session_id_when_unacknowledged_writes
    Execute_with_update_should_not_send_session_id_when_unacknowledged_writes

The remaining methods in this class that use Unacknowledged don't seem to have this problem because after executing the unacknowledged write they read the resulting collection thus blocking long enough for the unacknowledged writes to be executed before the test method finishes.

Reply:
done